mods/default/nodes.lua

```diff
@@ -530,9 +530,12 @@ minetest.register_node("default:snow", {
 	}),
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		local belowPos = pos
+		belowPos.y = belowPos.y -1
```

Comment:
Don't.
Either use `table.copy()` or `vector.new` to drop the reference.

Reply:
fixed! (and thanks, I started learning Lua yesterday)